daffodil-test/src/test/resources/org/apache/daffodil/section13/packed/packed.tdml

```diff
@@ -965,4 +966,88 @@
+    lengthUnits="bytes" binaryNumberRep="bcd" binaryNumberCheckPolicy="strict"/>
+    <xs:element name="bcdBigIntToLong" dfdl:lengthKind="delimited">
+      <xs:complexType>
+        <xs:sequence>
+          <xs:element name="uint01" type="xs:unsignedInt" dfdl:representation="binary" dfdl:length="1" />
+          <xs:element name="string01" type="xs:string" dfdl:encoding="US-ASCII" dfdl:length="{ xs:int(fn:ceiling(../ex:uint01)) }" />
+        </xs:sequence>
+      </xs:complexType>
+    </xs:element>
+  </tdml:defineSchema>
+
+  <!-- Daffodil 2961 -->
+  <tdml:parserTestCase name="bcdBigIntToLongExpr" root="bcdBigIntToLong" model="bcdBigIntToLongExprSch"
+    description="Verify that unsigned packed values in an expression work correctly">
+
+    <tdml:document>
+      <tdml:documentPart type="byte">0568656C6C6F</tdml:documentPart>
+    </tdml:document>
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <bcdBigIntToLong>
+          <uint01>5</uint01>
+          <string01>hello</string01>
+        </bcdBigIntToLong>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+  </tdml:parserTestCase>
+
+  <tdml:defineSchema name="packedDecimalRangeChecks">
+    <xs:include schemaLocation="/org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+    <dfdl:format ref="ex:GeneralFormat" lengthKind="explicit" encoding="X-DFDL-HEX-MSBF" occursCountKind="implicit"
+    textNumberCheckPolicy="strict" textNumberPadCharacter="0" textNumberJustification="right"
+    lengthUnits="bytes" binaryNumberRep="packed" binaryPackedSignCodes="C D F C" binaryNumberCheckPolicy="strict"/>
+
+    <xs:element name="uint01" type="xs:unsignedInt" dfdl:representation="binary" dfdl:length="2" />
+    <xs:element name="int01" type="xs:int" dfdl:representation="binary" dfdl:length="6" />
+
+  </tdml:defineSchema>
+
+  <tdml:parserTestCase name="packedNegativeUnsigned" root="uint01" model="packedDecimalRangeChecks"
+    description="Using BCD formatted hex to indicate an integer number">
+
+    <tdml:document>
+      <tdml:documentPart type="byte">123D</tdml:documentPart>
+    </tdml:document>
+
+    <tdml:errors>
+      <tdml:error>Parse Error</tdml:error>
+      <tdml:error>out of range for type</tdml:error>
+    </tdml:errors>
+    </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="packedIntTooLarge" root="int01" model="packedDecimalRangeChecks"
+    description="Using BCD formatted hex to indicate INT_MAX + 1">
+
+    <tdml:document>
+      <tdml:documentPart type="byte">02147483648C</tdml:documentPart>
+    </tdml:document>
+
+    <tdml:errors>
+      <tdml:error>Parse Error</tdml:error>
+      <tdml:error>Error in packed data</tdml:error>
+      <tdml:error>out of range for type</tdml:error>
+    </tdml:errors>
+    </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="packedIntMax" root="int01" model="packedDecimalRangeChecks"
+    description="Using BCD formatted hex to indicate INT_MAX + 1">
```

Comment:
It doesn't look like you added a junit test in the scala file for this test.

Also, I think the description should say INT_MAX instead of INT_MAX + 1.